torch/utils/data/dataloader.py

```diff
@@ -213,6 +215,9 @@ class DataLoader(Generic[_T_co]):
     .. warning:: See :ref:`reproducibility`, and :ref:`dataloader-workers-random-seed`, and
                  :ref:`data-loading-randomness` notes for random seed related questions.
 
+    .. warning:: Setting `in_order` to `False` can harm reproducibility and may lead to a skewed data
```

Comment:
One potential issue here that occurred to me: since work is continuously being distributed to workers in round-robin fashion, we can hit a scenario where one slow or blocked worker can end up holding all the tasks, blocking other tasks from being scheduled.

As a follow up, when in_order is False, we should also distribute work to workers to keep them balanced since reproducibility is already given up in these scenarios.

Reply:
Yep! That issue occurs currently with in_order=True I think (out of order tasks are buffered up instead of being given out)?
Would like this added as a warning, and fixed in a later PR, or fixed as part of this PR?
My _guess_ would be to track the number of tasks given to each worker and check [here](https://github.com/pytorch/pytorch/blob/main/torch/utils/data/dataloader.py#L1476) if it's < the prefetch factor.

Reply:
I think it'd be fine to do in a separate PR, given this is default-off, although I think we'd like it to land together in the next release

Reply:
Yep I'm very happy to do the follow up PR!